Compiler is not GNU or Flang! Aborting...

I am getting the above error when trying to add coverage flags to my project. It seems to be related to checking the compilers for each of the available languages.

Relevant CMake code:
```cmake
# Debug info to help with this issue
get_property(LANGUAGES GLOBAL PROPERTY ENABLED_LANGUAGES)
message("ALL LANGUAGES: ${LANGUAGES}")
foreach(LANG ${LANGUAGES})
    message("${LANG} compiler is \"${CMAKE_${LANG}_COMPILER_ID}\"")
endforeach()

# Error hit by this line
include(CodeCoverage)
```

Relevant output of CMake:
```
ALL LANGUAGES: ASM;C;CXX;NONE
ASM compiler is ""
C compiler is "Clang"
CXX compiler is "Clang"
NONE compiler is ""
CMake Error at lib/cmake-modules/CodeCoverage.cmake:159 (message):
  Compiler is not GNU or Flang! Aborting...
Call Stack (most recent call first):
  CMakeLists.txt:132 (include)
```

Please let me know if there's any more information I can provide

--- a/CodeCoverage.cmake
+++ b/CodeCoverage.cmake
@@ -153,6 +153,7 @@ endif() # NOT GCOV_PATH
 
 # Check supported compiler (Clang, GNU and Flang)
 get_property(LANGUAGES GLOBAL PROPERTY ENABLED_LANGUAGES)
+list(REMOVE_ITEM LANGUAGES NONE)
 foreach(LANG ${LANGUAGES})
   if("${CMAKE_${LANG}_COMPILER_ID}" MATCHES "(Apple)?[Cc]lang")
     if("${CMAKE_${LANG}_COMPILER_VERSION}" VERSION_LESS 3)

